src/components/Popular/MostRecent.tsx

```diff
@@ -0,0 +1,73 @@
+// src/components/MostPopular/MostPopular.tsx
+import { useState } from 'react';
+import { useSelector } from 'react-redux';
+import { RootState } from '../../app/store';
+import PopilarTitle from './PopilarTitle';
+import SingleItem from './Item';
+
+function MostRecent() {
+  const { availableProduct, status } = useSelector(
+    (state: RootState) => state.availableProducts
+  );
+
+  const [start, setStart] = useState(0);
+  const [end, setEnd] = useState(3);
+
+  const handleLeftallowclick = async () => {
+    if (start > 0) {
+      setEnd(end - 3);
+      setStart(start - 3);
+    } else {
+      setEnd(3);
+      setStart(0);
+    }
+  };
+
+  const handleRightallowclick = async () => {
+    if (end < availableProduct.length) {
+      setEnd(end + 3);
+      setStart(start + 3);
+    }
+  };
+
+  const mostRecentProducts = [...availableProduct]
+    .sort(
+      (a, b) =>
+        new Date(b.updatedAt).getTime() - new Date(a.createdAt).getTime()
+    )
+    .slice(start, end);
+
+  return (
+    <div className=" flex flex-col mb-6">
+      <PopilarTitle
+        section="Recent Products"
+        onLeftArrowClick={handleLeftallowclick}
+        onRightArrowClick={handleRightallowclick}
+      />
+
+      <div className=" grid gap-y-2">
+        {(status === 'failed' || status === 'loading') &&
```

Comment:
handle errors in other way and show the end users feedback about what is going on.

Reply:
done

Reply:
done